### PR TITLE
Misc updates

### PR DESCRIPTION
**Pull request by rustyrussell**
For routeboost testing (#1982), I wanted these, but they're really unrelated.

A "misc crap" PR like this is terrible practice, but separating them also leads to rebase hell...

### CODE REVIEW

File: wallet/test/run-wallet.c [hunk 1]
```diff
@@ -176,6 +176,10 @@ void invoices_waitone(const tal_t *ctx UNNEEDED,
 void json_add_bool(struct json_result *result UNNEEDED, const char *fieldname UNNEEDED,
 		   bool value UNNEEDED)
 { fprintf(stderr, "json_add_bool called!\n"); abort(); }
+/* Generated stub for json_add_hex */
+void json_add_hex(struct json_result *result UNNEEDED, const char *fieldname UNNEEDED,
```
Comment (niftynei) [inline wallet/test/run-wallet.c]: fancy. where do these generated stubs get generated?

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ### Added
 
 - JSON API: `listpeers` has new field `scratch_txid`: the latest tx in channel.
+- JSON API: `listpeers` has new array `htlcs`: the current live payments.
```
Comment (niftynei) [inline CHANGELOG.md]: :100:

**Comment by cdecker** [comment]
ACK https://github.com/ElementsProject/lightning/pull/1983/commits/83bc972c9ac77fb32053029b66a42a3a2bd222a5

File: wallet/test/run-wallet.c (hunk 1 above)
Comment (cdecker) [reply wallet/test/run-wallet.c]: `make update-mocks` will do that (but don't look behind the curtain) :wink:

**Pull request merged**